### PR TITLE
udev: use list.h instead of udev_list_node

### PR DESCRIPTION
**Pull request by keszybz**
This is a trivial fixup (some missing spaces) and rebase of #4600.

### CODE REVIEW

**Comment by poettering** [comment]
lgtm

File: src/udev/udevd.c [hunk 1]
```diff
@@ -160,8 +157,9 @@ static void event_free(struct event *event) {
 
         if (!event)
                 return;
+        assert(event->manager);
 
-        udev_list_node_remove(&event->node);
+        LIST_REMOVE(event,event->manager->events,event);
```
Comment (dnicolodi) [inline src/udev/udevd.c]: Missing space after `,`

Comment (poettering) [reply src/udev/udevd.c]: indeed

**Comment by keszybz** [comment]
Yikes. Updated.

**Comment by keszybz** [comment]
I'll restart amd64.

**Comment by keszybz** [comment]
```
TEST RUN: Basic systemd setup
+ /usr/bin/qemu-system-x86_64 -smp 1 -net none -m 512M -nographic -kernel /boot/vmlinuz-4.4.0-101-generic -drive format=raw,cache=unsafe,file=/var/tmp/systemd-test.bV7euy/rootdisk.img -initrd /boot/initrd.img-4.4.0-101-generic -machine accel=kvm -enable-kvm -cpu host -append 'ro root=/dev/sda1 raid=noautodetect loglevel=2 init=/lib/systemd/systemd console=ttyS0 selinux=0   '
KVM: entry failed, hardware error 0x0
EAX=00000000 EBX=00000000 ECX=00000000 EDX=000306c1
ESI=00000000 EDI=00000000 EBP=00000000 ESP=00000000
EIP=0000fff0 EFL=00000002 [-------] CPL=0 II=0 A20=1 SMM=0 HLT=0
ES =0000 00000000 0000ffff 00009300
CS =f000 ffff0000 0000ffff 00009b00
SS =0000 00000000 0000ffff 00009300
DS =0000 00000000 0000ffff 00009300
FS =0000 00000000 0000ffff 00009300
GS =0000 00000000 0000ffff 00009300
LDT=0000 00000000 0000ffff 00008200
TR =0000 00000000 0000ffff 00008b00
GDT=     00000000 0000ffff
IDT=     00000000 0000ffff
CR0=60000010 CR2=00000000 CR3=00000000 CR4=00000000
DR0=0000000000000000 DR1=0000000000000000 DR2=0000000000000000 DR3=0000000000000000 
DR6=00000000ffff0ff0 DR7=0000000000000400
EFER=0000000000000000
Code=ff ff 66 5b 66 83 c4 08 66 5b 66 5e 66 c3 b0 20 e6 20 66 c3 <ea> 5b e0 00 f0 30 36 2f 32 33 2f 39 39 00 fc 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
autopkgtest [13:28:21]: ERROR: timed out on command "su -s /bin/bash root -c set -e; export USER=`id -nu`; . /etc/profile >/dev/null 2>&1 || true;  . ~/.profile >/dev/null 2>&1 || true; buildtree="/tmp/autopkgtest.SMd7L6/build.fwq/debian"; mkdir -p -m 1777 -- "/tmp/autopkgtest.SMd7L6/upstream-artifacts"; export AUTOPKGTEST_ARTIFACTS="/tmp/autopkgtest.SMd7L6/upstream-artifacts"; export ADT_ARTIFACTS="$AUTOPKGTEST_ARTIFACTS"; mkdir -p -m 755 "/tmp/autopkgtest.SMd7L6/autopkgtest_tmp"; export AUTOPKGTEST_TMP="/tmp/autopkgtest.SMd7L6/autopkgtest_tmp"; export ADTTMP="$AUTOPKGTEST_TMP"; export DEBIAN_FRONTEND=noninteractive; export LANG=C.UTF-8; export DEB_BUILD_OPTIONS=parallel=1; unset LANGUAGE LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE   LC_MONETARY LC_MESSAGES LC_PAPER LC_NAME LC_ADDRESS   LC_TELEPHONE LC_MEASUREMENT LC_IDENTIFICATION LC_ALL;rm -f /tmp/autopkgtest_script_pid; set -C; echo $$ > /tmp/autopkgtest_script_pid; set +C; trap "rm -f /tmp/autopkgtest_script_pid" EXIT INT QUIT PIPE; cd "$buildtree"; export AUTOPKGTEST_NORMAL_USER=ubuntu; export ADT_NORMAL_USER=ubuntu; export 'CFLAGS=-O0'; export 'DEB_BUILD_PROFILES=noudeb'; export 'TEST_UPSTREAM=1'; export 'UPSTREAM_PULL_REQUEST=7398'; export 'GITHUB_STATUSES_URL=https://api.github.com/repos/systemd/systemd/statuses/bc9c9efd072ec553f198ae1c2a90014b88b354b8'; chmod +x /tmp/autopkgtest.SMd7L6/build.fwq/debian/debian/tests/upstream; touch /tmp/autopkgtest.SMd7L6/upstream-stdout /tmp/autopkgtest.SMd7L6/upstream-stderr; /tmp/autopkgtest.SMd7L6/build.fwq/debian/debian/tests/upstream 2> >(tee -a /tmp/autopkgtest.SMd7L6/upstream-stderr >&2) > >(tee -a /tmp/autopkgtest.SMd7L6/upstream-stdout);" (kind: test)
autopkgtest [13:28:21]: test upstream: -----------------------]
upstream             FAIL timed out
```

/cc @martinpitt, @xnox

**Comment by poettering** [comment]
would love to see the ubuntu amd64 CI to pass on this one before we merge it. I don't fully grok the effects of all of this, and the only integration tests we have for this are the qemu ones which only run on amd64... Can we restart the tests?

**Comment by keszybz** [comment]
The logs are here: https://objectstorage.prodstack4-5.canonical.com/v1/AUTH_77e2ada1e7a84929a74ba3b87153c0ac/autopkgtest-xenial-pitti-systemd-semaphore/xenial/amd64/s/systemd-upstream/20171122_134825_91e50@/log.gz, https://objectstorage.prodstack4-5.canonical.com/v1/AUTH_77e2ada1e7a84929a74ba3b87153c0ac/autopkgtest-artful-pitti-systemd-semaphore/artful/amd64/s/systemd-upstream/20171124_174343_91e50@/log.gz. I don't see why they fail. I'll restart.

**Comment by keszybz** [comment]
BTW, I restarted artful-amd64 once already. If it fails again, we shouldn't merge this without careful re-review.

**Comment by poettering** [comment]
artful-amd64 passed now, xenial-amd64 failed due to some CI-internal issue ("host testgreen…"). I figure this is good enough to merge now.

**Pull request merged**